srv/wordpress/wp-content/plugins/acore-wp-plugin/src/Components/AdminPanel/SettingsController.php

```diff
@@ -366,14 +366,88 @@ public function loadTools() {
             }
 
             // Reload configs
-            $this->data = $this->loadData(); 
+            $this->data = $this->loadData();
             ?>
                 <div class="updated"><p><strong>Tools have been saved</strong></p></div>
             <?php
         }
         echo $this->getView()->getToolsRender();
     }
 
+    public function loadSoapLogs() {
+        # defaults
+        $items = 10;
+        $pos = 1;
+        $userName = null;
+        $orderId = null;
+        $from = null;
+        $to = null;
+
+        if (!is_admin()) {
+            wp_die(__('You do not have sufficient permissions to access this page.'));
+        }
+        global $wpdb;
+        $soapLogsTableName = $wpdb->prefix . ACORE_SOAP_LOGS_TABLENAME;
+        $query = "SELECT
+            sl.*
+        FROM $soapLogsTableName sl
+        ##JOIN USERS##
+        WHERE
+            1=1
+            ##WHERE USERS##
+        ";
+        if (isset($_GET["username"]) && !empty($_GET["username"])) {
+            $userName = $_GET['username'];
+            $query = str_replace("##JOIN USERS##", "INNER JOIN {$wpdb->users} u ON sl.user_id = u.ID", $query);
+            $query = str_replace("##WHERE USERS##", " AND u.user_login LIKE '%{$wpdb->_real_escape($userName)}%'", $query);

```

Comment:
if you can, to improve a lot the efficiency of these queries move the WHERE conditions into the join, the join is computationally expensive, if you move some `where conditions` in the join you will save some CPU and memory.
ex.
```sql
INNER JOIN {$wpdb->users} u ON sl.user_id = u.ID AND u.user_login LIKE '%{$wpdb->_real_escape($userName)}%'"
```

this is not always possible but in this case it should be

Reply:
I'll change it.